test/cleanup.go

```diff
@@ -68,3 +69,16 @@ func TearDown(clients *Clients, names ResourceNames) {
 		)
 	}
 }
+
+// CleanupAfter will run the provided cleanup function when the test ends,
+// either via t.Cleanup or on interrupt via CleanupOnInterrupt
+func CleanupAfter(t *testing.T, cleanup func()) {
+	t.Cleanup(cleanup)
+	CleanupOnInterrupt(cleanup)
+}
+
+// TearDownAfter will delete created names when the test ends, either via
+// t.Cleanup, or on interrupt via CleanupOnInterrupt.
+func TearDownAfter(t *testing.T, clients *Clients, names ResourceNames) {
+	CleanupAfter(t, func() { TearDown(clients, names) })
```

Comment:
I'd drop `After`s :)

Reply:
yeah Im not loving the names here either but wanted to be clear that this does more than a regular Cleanup does. Maybe `EnsureCleanup` or `AlwaysCleanup`?

Reply:
EnsureCleanup — is better I think.